src/components/form/show.vue

```diff
@@ -122,55 +102,53 @@ export default {
           this.callWait(
             'fetchEnketoIdsForForm',
             async () => {
-              await this.$store.dispatch('get', [{
-                key: 'form',
+              await this.form.request({
                 url,
-                update: ['enketoId', 'enketoOnceId'],
+                patch: ({ data }) => {
+                  this.form.enketoId = data.enketoId;
+                  this.form.enketoOnceId = data.enketoOnceId;
+                },
                 alert: false
-              }]);
+              });
               return this.form.enketoId != null &&
                 this.form.enketoOnceId != null;
             },
             this.waitToRequestEnketoId
           );
-        }
-      }]).catch(noop);
+        })
+        .catch(noop);
     },
     fetchDraft() {
       this.cancelCall('fetchEnketoIdForDraft');
       const draftUrl = apiPaths.formDraft(this.projectId, this.xmlFormId);
-      this.$store.dispatch('get', [
-        {
-          key: 'formDraft',
-          url: draftUrl,
-          extended: true,
-          fulfillProblem: ({ code }) => code === 404.1,
-          success: () => {
+      Promise.allSettled([
+        this.formDraft.request({ url: draftUrl, extended: true })
+          .then(() => {
             if (this.formDraft.isEmpty()) return;
             if (this.formDraft.get().enketoId != null) return;
             this.callWait(
               'fetchEnketoIdForDraft',
               async () => {
-                await this.$store.dispatch('get', [{
-                  key: 'formDraft',
+                await this.formDraft.request({
                   url: draftUrl,
-                  update: ['enketoId'],
+                  patch: ({ data }) => {
+                    if (this.formDraft.isDefined())
+                      this.formDraft.get().enketoId = data.enketoId;
+                  },
                   alert: false
-                }]);
+                });
+                if (this.formDraft.isEmpty()) return true;
```

Comment:
I checked that if a `patch` callback throws an error, that won't break Frontend. I also wrote some related tests, for example, verifying that even if a `patch` callback throws an error, the state of the resource is cleaned up.

I remembered that I think I've actually encountered this case where `this.formDraft.isEmpty()` within the `patch` callback — so we probably should make some change here and not just revert. That said, I think the underlying cause is different from what I thought. It's not a race condition between two requests: a GET request for the draft and its Enketo ID; and another request that will delete the draft (either publishing or abandoning it). I still think that that's an edge case. Instead, I think it's more likely that the sequence I observed is that the GET request for the draft is sent _after_ the response to the request that deletes the draft. The GET request results in a 404.1 Problem, but that's processed as an expected response to the request, not an error, so the `patch` callback is called when it shouldn't be. I'm going to make two changes related to this:

1. Check before sending the GET request whether the draft has been deleted. That should be enough to handle this particular sequence. This can be seen in 2d09dcbc249f5e74156901a606bb38d551b62df0. (I also moved around some checks, which reduces duplication and I think is ultimately clearer.)
2. More broadly, this points to an issue with `Option` resources in this PR. On the `master` branch right now, the main request for `formDraft` specifies a `fulfillProblem` option that will accept a 404.1 Problem; but the request within the `callWait()` callback does not specify `fulfillProblem`. However, in this PR, I've set up `Option` resources so that they always specify `fulfillProblem`. I could override the `request()` method for `Option` resources so that they don't specify that default `fulfillProblem` option when a `patch` callback is specified. But that would be more complexity than I'm looking to add in this area and I think would bring up inheritance questions around resources, which I want to avoid. I'm going to go back to how things are on the `master` branch here: `Option` resources will not automatically specify `fulfillProblem`. That also means that I'll be able to remove the `requestDefaults` property. This change can be seen in c1e336e090e54a2f4f0a5547f59647c8a5112c08.

I think this might be an interesting area for further thought. For example, maybe we should automatically cancel a request for a resource if the resource's `data` property is set during the request? I'm going to keep some of the logic and the test for the edge case in case that's a useful example to consider as we continue thinking about this area.

Despite this long comment, all this doesn't amount to a huge change and will mostly be returning to how things already work on the `master` branch. Given that, I don't think that @ktuite and I necessarily need to review these commits. However, @ktuite, please ping me if you want to discuss any of this!